deps/wazuh_testing/wazuh_testing/tools/agent_simulator.py

```diff
@@ -0,0 +1,806 @@
+        context.verify_mode = ssl.CERT_NONE
+        ssl_socket = context.wrap_socket(sock,
+                                         server_hostname=self.manager_address)
+        ssl_socket.connect((self.manager_address, 1515))
+        if self.authd_password is None:
+            event = "OSSEC A:'{}'\n".format(self.name).encode()
+        else:
+            event = "OSSEC PASS: {} OSSEC A:'{}'\n".format(self.authd_password,
+                                                           self.name).encode()
+        ssl_socket.send(event)
+        recv = ssl_socket.recv(4096)
+        registration_info = recv.decode().split("'")[1].split(" ")
+        self.id = registration_info[0]
+        self.key = registration_info[3]
+        ssl_socket.close()
+        sock.close()
+        print("Registration - {}({})".format(self.name, self.id))
+
+    # Add the Wazuh custom padding to each event sent
+    @staticmethod
+    def wazuh_padding(compressed_event):
+        padding = 8
+        extra = len(compressed_event) % padding
+        if extra > 0:
+            padded_event = (b'!' * (padding - extra)) + compressed_event
+        else:
+            padded_event = (b'!' * padding) + compressed_event
+        return padded_event
+
+    # Generate encryption key (using agent metadata and key)
+    def create_encryption_key(self):
+        id = self.id.encode()
+        name = self.name.encode()
+        key = self.key.encode()
+        sum1 = (hashlib.md5((hashlib.md5(name).hexdigest().encode()
+                + hashlib.md5(id).hexdigest().encode())).hexdigest().encode())
+        sum1 = sum1[:15]
+        sum2 = hashlib.md5(key).hexdigest().encode()
+        key = sum2 + sum1
+        self.encryption_key = key
+
+    # Compose event from raw message
+    @staticmethod
+    def compose_event(message):
+        message = message.encode()
+        random_number = b'55555'
+        global_counter = b'1234567891'
+        split = b':'
+        local_counter = b'5555'
+        msg = random_number + global_counter + split + local_counter \
+            + split + message
+        msg_md5 = hashlib.md5(msg).hexdigest()
+        event = msg_md5.encode() + msg
+        return event
+
+    # Encrypt event AES or Blowfish
+    def encrypt(self, padded_event):
+        if self.cypher == "aes":
+            encrypted_event = Cipher(padded_event,
+                                     self.encryption_key).encrypt_aes()
+        if self.cypher == "blowfish":
+            encrypted_event = Cipher(padded_event,
+                                     self.encryption_key).encrypt_blowfish()
+        return encrypted_event
+
+    # Add event headers for AES or Blowfish Cyphers
+    def headers(self, agentid, encrypted_event):
+        if self.cypher == "aes":
+            header = "!{0}!#AES:".format(agentid).encode()
+        if self.cypher == "blowfish":
+            header = "!{0}!:".format(agentid).encode()
+        headers_event = header + encrypted_event
```

Comment:
Local variable 'header' might be referenced before assignment, if neither of the above conditions are satisfied